core/impl/cluster.cxx

```diff
@@ -297,11 +300,13 @@ cluster::query(std::string statement, const query_options& options, query_handle
 }
 
 auto
-cluster::query(std::string statement, const query_options& options) const -> std::future<std::pair<query_error_context, query_result>>
+cluster::query(std::string statement, const query_options& options) const -> std::future<std::pair<error, query_result>>
 {
-    auto barrier = std::make_shared<std::promise<std::pair<query_error_context, query_result>>>();
+    auto barrier = std::make_shared<std::promise<std::pair<error, query_result>>>();
     auto future = barrier->get_future();
-    query(std::move(statement), options, [barrier](auto ctx, auto result) { barrier->set_value({ std::move(ctx), std::move(result) }); });
+    query(std::move(statement), options, [barrier](auto ctx, auto result) {
```

Comment:
It would be good to rename any references to `ctx` where it is a `couchbase::error` to `err`